### PR TITLE
Update error codes related to tasks cancelation + add canceledBy filter

### PR DESCRIPTION
**Pull request by Kerollmops**
<details>This PR changes the error codes [to follow the specification](https://github.com/meilisearch/specifications/pull/195).

 - [x] The `missing_filters` error code is renamed `missing_task_filters` to be more accurate and follow the `invalid_task_*` convention.
 - [x] The error code `invalid_task_uids_filter` is added.
 - [x] The error code `invalid_task_canceled_by_filter` is added.
 - [x] The error code `invalid_task_date_filter` is added.
      -  The error message is the same as for expires_at in the API Key  EXCEPT that it does not explicitly mention that a date must be given in the future.
</details>

Edit by @loiclec :
I have added a few more changes into this PR. The related issues are:

- Fixes https://github.com/meilisearch/meilisearch/issues/3029
- Implements https://github.com/meilisearch/meilisearch/issues/3026
- Fixes https://github.com/meilisearch/meilisearch/issues/2940
- Fixes https://github.com/meilisearch/meilisearch/issues/2939

Additionally:
- Fixes a bug where global tasks were returned by `GET /tasks` queries even if the user did not have the `index.*` API key action.
- Rename `originalQuery` to `originalFilters`
- Display `error: null` and `canceledBy: null` in the task views
- Allow using the star operator in the task filters in the `DELETE /tasks` and `POST /tasks/cancel` routes
- Make sure that the index scheduler keeps making progress even when a grave error occurs.

### CODE REVIEW

**Comment by curquiza** [comment]
I assign @loiclec here, because he's going to try to continue the work (discussed with @Kerollmops who made me aware of the decision)

**Comment by gmourier** [comment]
I've updated the PR description to reflect the latest changes regarding the error code names following https://github.com/meilisearch/meilisearch/issues/3029

File: index-scheduler/src/lib.rs [hunk 1]
```diff
@@ -422,6 +430,7 @@ impl IndexScheduler {
                     ) {
                         std::thread::sleep(Duration::from_secs(1));
                     }
+                    run.wake_up.signal();
```
Comment (Kerollmops) [inline index-scheduler/src/lib.rs:433]: Why do we have to signal the loop, here? Is it a bug fix?

File: index-scheduler/src/lib.rs [hunk 2]
```diff
@@ -256,6 +260,9 @@ pub struct IndexScheduler {
     /// Store the tasks associated to an index.
     pub(crate) index_tasks: Database<Str, RoaringBitmapCodec>,
 
+    /// Store the tasks that were canceled by a task uid
+    pub(crate) canceled_by: Database<OwnedType<BEU32>, RoaringBitmapCodec>,
```
Comment (Kerollmops) [inline index-scheduler/src/lib.rs:264]: Nice, you added a database where the key is the task-cancelation id, and the value is a roaring bitmap with the tasks it canceled. That's cool ❤️

File: index-scheduler/src/batch.rs [hunk 3]
```diff
@@ -1222,6 +1226,17 @@ impl IndexScheduler {
...
+            let canceled_by = BEU32::new(canceled_by);
+            if let Some(mut tasks) = self.canceled_by.get(wtxn, &canceled_by)? {
+                tasks -= &to_delete_tasks;
+                if tasks.is_empty() {
+                    self.canceled_by.delete(wtxn, &canceled_by)?;
+                } else {
+                    self.canceled_by.put(wtxn, &canceled_by, &tasks)?;
+                }
+            }
+        }
```
Comment (Kerollmops) [inline index-scheduler/src/batch.rs:1239]: Ok so here you clean up the deleted tasks from the lists of canceled tasks, right?

File: index-scheduler/src/lib.rs (hunk 1 above)
Comment (loiclec) [reply index-scheduler/src/lib.rs:433]: Yes it is a bug fix, but I didn't remember leaving it there. Basically we want the index scheduler to keep making progress even when a grave error occurs. Do you want it in a separate PR?

File: index-scheduler/src/lib.rs (hunk 2 above)
Comment (loiclec) [reply index-scheduler/src/lib.rs:264]: haha, yes you were completely right about needing another database :)

File: index-scheduler/src/lib.rs (hunk 1 above)
Comment (Kerollmops) [reply index-scheduler/src/lib.rs:433]: No, that's ok but can you add it in the PR description please?

**Comment by curquiza** [comment]
@Kerollmops or @loiclec could you add all the related issues to the `Development` section of this PR please? 😇

**Review by loiclec — APPROVED** [review]

**Comment by Kerollmops** [comment]
bors merge

**Pull request merged**